File: 0050_Pow(x, n)/Pow(x, n)_Recursion.cs
public class Solution {
    public double MyPow(double x, int n) {
        if(n < 0) {
            x = 1 / x;
            n = -n;
        }
        
        return calc(x, n);
    }
    
    private double calc(double x, int n) {
        if(n == 0) return 1.0;
        
        double half = calc(x, n/2);
        if(n % 2 == 0) 
            return half * half;
        else
            return half * half * x;
    }
}

File: 0050_Pow(x, n)/Pow(x, n)_Recursion.java
class Solution {
    public double myPow(double x, int n) {
        return pow(x, n);
    }
    
     private double pow(double x, long n) {
        if(n < 0) return 1 / pow(x, -n);
        if(n == 0) return 1.0;
        if(n % 2 == 0) {
            double v = pow(x, n/2);
            return v * v;
        }else{
            return x * pow(x, n-1);
        }
    }
}